### ISSUE TITLE
Marking attributes as immutable, updatable, and creatable

### ISSUE THREAD

**Issue by warmwaffles** [JSONAPI-Resources/jsonapi-resources#793]
It would be nice to be able to specify an attribute that it can't be updated.

``` ruby
class SomeResource < ApplicationResource
  key_type :uuid
  attribute :foo, updatable: false
  attribute :created_at, immutabe: true
  attribute :updated_at, immutable: true
end
```

If this was a PR, would it be acceptable?

**Comment by lgebhardt** [comment]
It seems useful to me.

**Comment by olegantonyan** [comment]
Hi @warmwaffles are there any news on this PR? Are you already working on it or should I start implementing this?

**Comment by mroach** [comment]
It seems like this is already supported, albeit with a different approach (method vs attributes): https://github.com/cerebris/jsonapi-resources#creatable-and-updatable-attributes

**Comment by olegantonyan** [comment]
@mroach yes, this should solve the problem. though I don't like the api, declarative attributes are better imo

**Comment by mroach** [comment]
@olegantonyan They are nice, and perhaps they could be one of two options for implementing?

The reason I'm a fan of the method approach is it would allow you to load the fields from somewhere else. For example, Pundit's [`permitted_attributes_for_update`](https://github.com/elabs/pundit#strong-parameters). Although since these are implemented as class methods in JR, it's not possible to do this with the current implementation.

Example case: A user has permission to update their own name and email address, but not anyone else's. An admin has permission to update anyone's name, address, email, _and_ security roles. As I see it, the only way to do this with the current implementation would be to override a bunch of stuff in the controllers.

**Comment by olegantonyan** [comment]
@mroach yep, this makes sense. Agreed

It feels to me that https://github.com/venuu/jsonapi-authorization should handle pundit's permitted attributes automagically. I have to invest more time into this issue.

**Comment by warmwaffles** [comment]
> Hi @warmwaffles are there any news on this PR? Are you already working on it or should I start implementing this?

I have not started on it @olegantonyan. There are some attributes that just wont be editable even if permissions are high enough. `created_at`, `updated_at`, other system generated values that can not be supplied by the user.
